src/WorkSpace.ts

```diff
@@ -0,0 +1,38 @@
+import path from 'node:path'
+import compact from 'lodash.compact'
+
+import FileObject from './FileObject'
+
+export default class WorkSpace {
+  name: string
+  fileUri: string
+  uri: string
+
+  constructor(fileUri: string) {
+    this.fileUri = fileUri
+  }
+
+  toJSON() {
+    const fileUriArray = compact(this.fileUri.split(path.sep))
+
+    const findIndexCallback = (o) => new RegExp(`^(app|spec|test|lib)$`).test(o)
+    const index = fileUriArray.findIndex(findIndexCallback)
+    const array = fileUriArray.slice(0, index)
+
+    this.uri = array.slice(0, index).join('/')
```

Comment:
detected duplicate slice

```suggestion
    this.uri = array.join('/')
```